x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsRepositoryIntegTests.java

```diff
@@ -110,4 +117,130 @@ public void testRepositoryUsedBySearchableSnapshotCanBeUpdatedButNotUnregistered
+                containsString(':' + snapshot + "] as index [" + mountedAgain + "] with "),
+                containsString("[index.store.snapshot.delete_searchable_snapshot: " + (deleteSnapshot == false) + "]; another "),
+                containsString("index [" + mounted + '/'),
+                containsString("is mounted with [index.store.snapshot.delete_searchable_snapshot: " + deleteSnapshot + "].")
+            )
+        );
+
+        mountSnapshot(repository, snapshot, index, mountedAgain, deleteSnapshotIndexSettings(deleteSnapshot));
+        assertIndexSetting(mountedAgain, SEARCHABLE_SNAPSHOTS_DELETE_SNAPSHOT_ON_INDEX_DELETION, equalTo(Boolean.toString(deleteSnapshot)));
+        assertHitCount(client().prepareSearch(mountedAgain).setTrackTotalHits(true).get(), totalHits.value);
+
+        assertAcked(client().admin().indices().prepareDelete(mountedAgain));
+        assertAcked(client().admin().indices().prepareDelete(mounted));
+    }
+
+    public void testDeletionOfSnapshotSettingCannotBeUpdated() throws Exception {
+        final String repository = "repository-" + getTestName().toLowerCase(Locale.ROOT);
+        createRepository(repository, FsRepository.TYPE, randomRepositorySettings());
+
+        final String index = "index";
+        createAndPopulateIndex(index, Settings.builder().put(INDEX_SOFT_DELETES_SETTING.getKey(), true));
+
+        final TotalHits totalHits = internalCluster().client().prepareSearch(index).setTrackTotalHits(true).get().getHits().getTotalHits();
+
+        final String snapshot = "snapshot";
+        createSnapshot(repository, snapshot, List.of(index));
+        assertAcked(client().admin().indices().prepareDelete(index));
+
+        final String mounted = "mounted-" + index;
+        final boolean deleteSnapshot = randomBoolean();
+
+        mountSnapshot(repository, snapshot, index, mounted, deleteSnapshotIndexSettings(deleteSnapshot), randomFrom(Storage.values()));
+        assertIndexSetting(mounted, SEARCHABLE_SNAPSHOTS_DELETE_SNAPSHOT_ON_INDEX_DELETION, equalTo(Boolean.toString(deleteSnapshot)));
+        assertHitCount(client().prepareSearch(mounted).setTrackTotalHits(true).get(), totalHits.value);
+
+        final IllegalArgumentException exception = expectThrows(
+            IllegalArgumentException.class,
+            () -> client().admin()
+                .indices()
+                .prepareUpdateSettings(mounted)
+                .setSettings(deleteSnapshotIndexSettings(deleteSnapshot == false))
+                .get()
+        );
+        assertThat(
+            exception.getMessage(),
+            containsString("can not update private setting [index.store.snapshot.delete_searchable_snapshot]; ")
+        );
+
+        assertAcked(client().admin().indices().prepareDelete(mounted));
+    }
+
+    private static Settings deleteSnapshotIndexSettings(boolean value) {
+        return Settings.builder().put(SEARCHABLE_SNAPSHOTS_DELETE_SNAPSHOT_ON_INDEX_DELETION, value).build();
```

Comment:
Is having to use this setting in the API an interim state, with a follow-up PR adding a direct flag in the API?

Otherwise, I think the setting should not be private if the intention is that external users should set this flag using the mount call.

Reply:
> Is having to use this setting in the API an interim state, with a follow-up PR adding a direct flag in the API?

I planned to use the setting in the Mount API and not introducing a flag in the API.

I still think it can be private - in a follow up I'd like to always set this setting on indices mounted in 8.0+.